api/base/parsers.py

```diff
@@ -15,6 +17,29 @@ class JSONAPIParser(JSONParser):
     media_type = 'application/vnd.api+json'
     renderer_class = JSONAPIRenderer
 
+    def flatten_relationships(self, relationships):
+        """
+        Flattens relationships dictionary which has information needed to create related resource objects.
+
+        Validates that formatting of relationships dictionary is correct.
+        """
+        if not isinstance(relationships, dict):
+            raise ParseError()
+        related_resource = relationships.keys()[0]
```

Comment:
That's a very meaningful assumption. If we ever decide to have more than one relationship, this is going to be a very painful bug to track down.

If it's not an easy fix based on the structure of data you operate on, then I can see the case for fixing that in a followon PR. But at absolute minimum there should be a highly visible note warning future developers. (and quite possibly a ticket linked to this issue)